pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,418 @@
+from collections import defaultdict
+from ctypes import (c_char_p, c_double, c_float, c_long, c_longlong, c_void_p,
+                    RTLD_GLOBAL)
+import re
+
+import numpy as np
+
+from pyfr.ctypesutil import LibWrapper
+from pyfr.mpiutil import get_comm_rank_root
+from pyfr.nputil import npeval
+from pyfr.plugins.base import BasePlugin, RegionMixin
+from pyfr.shapes import BaseShape
+from pyfr.util import file_path_gen, subclass_where
+from pyfr.writers.vtk import BaseShapeSubDiv
+
+
+def get_expr(expr, subs={}):
```

Comment:
I'd _ this.  Also, do not need to make subs optional (we always pass them) which will get rid of the if statement.